src/hdf5/include/H5public.h

```diff
@@ -338,10 +338,15 @@ H5_DLL herr_t H5get_libversion(unsigned *majnum, unsigned *minnum,
 H5_DLL herr_t H5check_version(unsigned majnum, unsigned minnum,
 			       unsigned relnum);
 H5_DLL herr_t H5is_library_threadsafe(hbool_t *is_ts);
-H5_DLL herr_t H5free_memory(void *mem);
-H5_DLL void *H5allocate_memory(size_t size, hbool_t clear);
+// Special case
+// H5_DLL herr_t H5free_memory(void *mem);
+// H5_DLL void *H5allocate_memory(size_t size, hbool_t clear);
 H5_DLL void *H5resize_memory(void *mem, size_t size);
 
+// Filters should use malloc/free
+#define H5free_memory(mem) free(mem)
+#define H5allocate_memory(size, clear) malloc(size)
+

```

Comment:
The new version of the LZ4 filter use H5allocate_memory/H5free_memory.
From what I understand this should not be used here in filters. No other filter is using this, even filters in https://github.com/HDFGroup/hdf5_plugins/ use malloc/free.

So I defined macros to use malloc/free instead.